File: app/src/main/java/com/boardgamegeek/sorter/PlayersWinSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.database.Cursor;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;
import com.boardgamegeek.provider.BggContract.Plays;

public class PlayersWinSorter extends PlayersSorter {
	public PlayersWinSorter(@NonNull Context context) {
		super(context);
		orderByClause = getClause(Plays.SUM_WINS, true);
	}

	@StringRes
	@Override
	protected int getDescriptionId() {
		return R.string.menu_sort_wins;
	}

	@Override
	public int getType() {
		return PlayersSorterFactory.TYPE_WINS;
	}

	@Override
	public String[] getColumns() {
		return new String[] { Plays.SUM_WINS };
	}

	@NonNull
	@Override
	public String getHeaderText(@NonNull Cursor cursor) {
		int q = getInt(cursor, Plays.SUM_WINS);
		String prefix = String.valueOf(q).substring(0, 1);
		String suffix = "";
		if (q >= 10000) {
			suffix = "0000+";
		} else if (q >= 1000) {
			suffix = "000+";
		} else if (q >= 100) {
			suffix = "00+";
		} else if (q >= 10) {
			suffix = "0+";
		}
		return prefix + suffix;
	}

	@Override
	public String getDisplayInfo(Cursor cursor) {
		int winCount = getInt(cursor, Plays.SUM_WINS);
		return context.getResources().getQuantityString(R.plurals.wins_suffix, winCount, winCount);
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayersWinSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import android.database.Cursor
import androidx.annotation.StringRes

import com.boardgamegeek.R
import com.boardgamegeek.extensions.getInt
import com.boardgamegeek.provider.BggContract.Plays

class PlayersWinSorter(context: Context) : PlayersSorter(context) {

    override val descriptionResId: Int
        @StringRes
        get() = R.string.menu_sort_wins

    override val type: Int
        get() = PlayersSorterFactory.TYPE_WINS

    override val sortColumn: String
        get() = Plays.SUM_WINS

    override val isSortDescending: Boolean
        get() = true

    public override fun getHeaderText(cursor: Cursor): String {
        val quantity = cursor.getInt(Plays.SUM_WINS)
        val prefix = quantity.toString().substring(0, 1)
        val suffix = when {
            quantity >= 10000 -> "0000+"
            quantity >= 1000 -> "000+"
            quantity >= 100 -> "00+"
            quantity >= 10 -> "0+"
            else -> ""
        }
        return prefix + suffix
    }

    override fun getDisplayInfo(cursor: Cursor): String {
        val winCount = cursor.getInt(Plays.SUM_WINS)
        return context.resources.getQuantityString(R.plurals.wins_suffix, winCount, winCount)
    }
}
